subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -517,6 +526,13 @@ static int hawkbit_parse_deployment(struct hawkbit_dep_res *res,
 	}
 
 	artifact = &chunk->artifacts[0];
+
+	if (hex2bin(artifact->hashes.sha256, strlen(artifact->hashes.sha256),
```

Comment:
For safety reasons, I would consider using `strnlen` or eventually `mbedtls_md_get_size(hash_info)` as size reference:
```c
size_t hash_size = mbedtls_md_get_size(hash_info);
```

Reply:
Make sense, edited